node.gypi

```diff
@@ -41,6 +41,19 @@
       'defines': [
         'NODE_SHARED_MODE',
       ],
+      'conditions': [
```

Comment:
Is this necessary for all platforms? I'm wondering since the only issues you mentioned for linux/osx don't seem related to this.

Reply:
I think you may be right that these are only necessary on Windows. Their purpose is to ensure that libuv and v8 classes are correctly annotated with `__declspec(dllimport)` rather than `__declspec(dllexport)`. I will refactor this condition block to only apply to Windows.

Reply:
It shouldn't hurt (and is more correct) to enable these on non-Windows.